bootstrap.sh

```diff
@@ -65,133 +75,228 @@ PermitRootLogin no
 UsePrivilegeSeparation sandbox
 Subsystem sftp /usr/libexec/sftp-server
 EOF
-  
+
+}
+
+dot_bashrc_setup() {
+
   echo "--- Overwrite .bashrc with our own."
   cat > ~/.bashrc <<EOF
+# generated by travis
 export LC_ALL="en_US.UTF-8"
 export LANG="en_US.UTF-8"
 export TRAVIS=true
 export CI=true
 EOF
-  
+}
+
+dot_profile_setup() {
   echo "--- Ensure that ~/.profile loads ~/.bashrc"
   cat > ~/.profile <<EOF
+# generated by travis
 [[ -s "\$HOME/.bashrc" ]] && source "\$HOME/.bashrc"
 EOF
-  
+}
+
+dot_bash_profile_setup() {
   echo "--- Ensure that ~/.bash_profile contents are correct"
   cat > ~/.bash_profile <<EOF
+# generated by travis
+
+# load .profile
 [[ -s "\$HOME/.profile" ]] && source "\$HOME/.profile"
-[[ -s "\$HOME/.rvm/scripts/rvm" ]] && source "\$HOME/.rvm/scripts/rvm"
+
+# help nvm work
 export NVM_DIR="\$HOME/.nvm"
 [[ -s "\$NVM_DIR/nvm.sh" ]] && source "\$NVM_DIR/nvm.sh"
+
+# help rvm work
+[[ -s "\$HOME/.rvm/scripts/rvm" ]] && source "\$HOME/.rvm/scripts/rvm"
+export PATH="\$PATH:$HOME/.rvm/bin"
 EOF
-  
-  echo "--- add 'gem: --no-document' so gem installs don't include documentation"
-  cat > ~/.gemrc <<EOF
+  # we want what we've written to take effect before we do more work
+  source ~/.bash_profile
+}
+
+gemrc_setup() {
+  if [[ ! $(grep no-document ~/.gemrc) ]]; then
+    echo "--- add 'gem: --no-document' so gem installs don't include documentation"
+    cat > ~/.gemrc <<EOF
 gem: --no-document
 EOF
-  
-  echo "--- Turn off automatic software updating"
-  sudo softwareupdate --schedule off
-  
+  else
+    echo "--- .gemrc is good. Skipping."
+  fi
+}
+
+disable_scheduled_software_updates() {
+  if [[ ! $(sudo softwareupdate --schedule | grep off) ]]; then
```

Comment:
I don't trust OS X to not randomly turn this back on after system updates, so it'd be nice to run this every time.